fendermint/vm/topdown/src/sync.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! A constant running process that fetch or listener to parent state
+
+use crate::error::Error;
+use crate::{BlockHash, BlockHeight, Config, InMemoryFinalityProvider, ParentFinalityProvider};
+use anyhow::Context;
+use async_stm::atomically_or_err;
+use fvm_shared::clock::ChainEpoch;
+use ipc_agent_sdk::apis::IpcAgentClient;
+use ipc_agent_sdk::jsonrpc::JsonRpcClientImpl;
+use ipc_sdk::cross::CrossMsg;
+use ipc_sdk::subnet_id::SubnetID;
+use ipc_sdk::ValidatorSet;
+use std::sync::Arc;
+use std::time::Duration;
+
+/// Constantly syncing with parent through polling
+pub struct PollingParentSyncer {
+    config: Config,
+    parent_view_provider: Arc<InMemoryFinalityProvider>,
+    agent: Arc<IPCAgentProxy>,
+}
+
+impl PollingParentSyncer {
+    pub fn new(
+        config: Config,
+        parent_view_provider: Arc<InMemoryFinalityProvider>,
+        agent: Arc<IPCAgentProxy>,
+    ) -> Self {
+        Self {
+            config,
+            parent_view_provider,
+            agent,
+        }
+    }
+}
+
+impl PollingParentSyncer {
+    /// Start the parent finality listener in the background
+    pub fn start(self) -> anyhow::Result<()> {
+        let config = self.config;
+        let provider = self.parent_view_provider;
+        let agent = self.agent;
+
+        tokio::spawn(async move {
+            loop {
+                if let Err(e) = sync_with_parent(&config, &agent, &provider).await {
+                    tracing::info!("sync with parent encountered error: {e}");
+                }
+            }
+        });
+
+        Ok(())
+    }
+}
+
+/// Extract the actual error from Box
+macro_rules! downcast_err {
+    ($r:expr) => {
+        match $r {
+            Ok(v) => Ok(v),
+            Err(e) => match e.downcast_ref::<Error>() {
+                None => unreachable!(),
+                Some(e) => Err(e.clone()),
+            },
+        }
+    };
+}
+
+async fn sync_with_parent<T: ParentFinalityProvider + Send + Sync + 'static>(
```

Comment:
Would you mind adding a comment on what is the end-to-end process so it is clear:
1. Get the heights to be populated in the parent view. 
2. Populate the finality information for each of these heights.